### PR TITLE
[Refactoring] Export student submissions

### PR DESCRIPTION
**Pull request by ivanchimeno**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I documented the Java code using JavaDoc style.

### Motivation and Context

At the moment, there is code duplication within the export student submissions functions for Text-, File-, and Modeling exercises.

### Description

This PR refactors the code and removes the duplicated code.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Generate a Text-, Modeling-, and File Upload Exercise
4. Submit to all three exercises
5. Navigate to the detail views of the exercises and press "export submissions"
6. Download the zip file, open it, and make sure that the submitted answer is inside

### Test Coverage

FileService.java 82%
SubmissionExportService.java 89%
FileUploadSubmissionResource.java 80%
ModelingExerciseResource.java 87%
TextExerciseResource.java 90%
ResponseUtil.java 94%

### CODE REVIEW

**Review by maximilianbergmann — APPROVED** [review]
Tested on TS3. Works as expected

**Review by real-dmy — APPROVED** [review]
Tested on TS1. Works as expected.

**Review by rs-maker — APPROVED** [review]
Tested on TS1 worked as expected. Code looks good too.

**Review by coolchock — APPROVED** [review]
tested on ts3, works as expected :+1:

**Review by jhainzlschmid — APPROVED** [review]
Tested on ts1 and worked. Code also looks good. Would be nice if you could fix the codacy issues about the static import, but it isn't important, so I am approving regardless

**Review by KiraNickel — APPROVED** [review]
Tested on ts3, works as described

**Review by sinangunduz — APPROVED** [review]
tested on ts1, works as expected

File: src/main/java/de/tum/in/www1/artemis/service/SubmissionExportService.java [hunk 1]
```diff
@@ -42,6 +48,42 @@ public SubmissionExportService(ExerciseRepository exerciseRepository, ZipFileSer
         this.fileService = fileService;
     }
 
+    /**
+     * Exports student submission of an exercise to a zip file located in the submission exports folder.
```
Comment (FuchsDominik) [inline src/main/java/de/tum/in/www1/artemis/service/SubmissionExportService.java:52]:
```suggestion
     * Exports student submissions of an exercise to a zip file located in the submission exports folder.
```

File: src/main/java/de/tum/in/www1/artemis/service/SubmissionExportService.java [hunk 2]
```diff
@@ -42,6 +48,42 @@ public SubmissionExportService(ExerciseRepository exerciseRepository, ZipFileSer
...
+     * @param submissionExportOptions the options for the export
+     * @return the zippped file with the exported submissions
+     */
+    public File exportStudentSubmissionsElseThrow(Long exerciseId, SubmissionExportOptionsDTO submissionExportOptions) {
+        return exportStudentSubmissions(exerciseId, submissionExportOptions)
+                .orElseThrow(() -> new BadRequestAlertException("Failed to export student submissions.", "SubmissionExport", "nosubmissions"));
+    }
+
+    /**
+     * Exports student submission of an exercise to a zip file located in the submission exports folder.
```
Comment (FuchsDominik) [inline src/main/java/de/tum/in/www1/artemis/service/SubmissionExportService.java:66]:
```suggestion
     * Exports student submissions of an exercise to a zip file located in the submission exports folder.
```

**Review by FuchsDominik — APPROVED** [review]
Code looks good

**Pull request merged**